cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -726,7 +761,7 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         }
 
         Future<ReconciliationState> kafkaAncillaryCm() {
-            return withKafkaAncillaryCmChanged(configMapOperations.reconcile(namespace, kafkaCluster.getAncillaryConfigName(), kafkaMetricsAndLogsConfigMap));
+            return getReconciliationStateOfConfigMap(kafkaCluster, kafkaMetricsAndLogsConfigMap);
```

Comment:
`getReconciliationStateOfConfigMap()` called `withZkAncillaryCmChanged()` which modifies the state (`zkForcedRestart`) of the `ReconciliationState` for something ZK-specific, but here you're calling it with something Kafka-related. I think that's must be incorrect.

Rather than duplicating `getReconciliationStateOfConfigMap()` (one for ZK and one for Kafka) it would be better to abstract it somehow. For example the caller could pass a thing (such as a method reference to `withZkAncillaryCmChanged`/``withKafkaAncillaryCmChanged`) which sets the field indirectly.